### PR TITLE
[iOS] Fix navigation bar background color in iOS13

### LINKED ISSUE #7350: [Bug] Large Titles not working in iOS 13
### Description

Take Xamarin.Forms sample, add the large titles to nav page, fails to show on iOS 13

https://github.com/xamarin/xamarin-forms-samples/tree/master/Navigation/TabbedPageWithNavigationPage

1. Set the navigation page to prefer large titles
2. Set the schedulepage.xaml to have large, set the upcomingappointmentspage.xaml to never show large
3. Run on iOS 12 and iOS 13 to compare

iOS 13 should behave as 12 and show titles as expected

iOS 13 shows nothing in navigation bar. 

### Basic Information

- Version with issue: 4.2.0.709249+ios13
- Last known good version:4.2.0.709249+ios12
- IDE:vs4mac
- Platform Target Frameworks: 
  - iOS:  13
- Nuget Packages:
- Affected Devices:all ios

### Screenshots

Short video of issue
https://www.dropbox.com/s/0ptzu3ychwinnk8/nolargetitles.mov?dl=0

### LINKED ISSUE #7679: [Bug] iOS large title doesn't clear background
When a `NavigationPage` has large titles enabled (`SetPrefersLargeTitles(navigationPage.On<iOS>(), true)`), the page transition animation shows the large title background as gray. After the navigation completes, the background changes to white. Additionally, if the prior page has a `ListView` with groups, the group headers continue to be visible on top of the large title during the animation. Each visual artifact makes the transition look janky.

This is a regression. I first encountered this bug after upgrading to XF 4.2.0.815419 and iOS 13.

### Basic Information

- Version with issue: 4.2.0.815419
- Last known good version: 4.0?
- IDE: VS 16.3.0 (Windows), VS 3.8 (build 1805) (Mac), Xcode 11
- Platform Target Frameworks:
  - iOS: 13.1

### Reproduction

ControlGallery => NavigationBar Gallery -Legacy => Go to Searchbar TitlePage

### LINKED ISSUE #7701: [Bug] iOS Large SetPrefersLargeTitles(true) Uses Incorrect BarBackgroundColor
### Description

When using `NavigationPage.On<iOS>().SetPrefersLargeTitles(true)`, the incorrect `BarBackgroundColor` is displayed.

### Steps to Reproduce

Reproduction Sample: https://github.com/brminnick/iOSLargeTitleRepro

1. Download the [reproduction sample](https://github.com/brminnick/iOSLargeTitleRepro)
2. Open `iOSLargeTitleRepro.sln` 
3. Build/deploy to device/simulator running iOS 13

### Expected Behavior

The `NavigationPage.BarBackgroundColor` should be `Color.Green` (as set in `App.cs`).

Note that when the page is scrolled, and the Navigation Bar is reduced, the correct color is displayed.

### Actual Behavior

The `NavigationPage.BarBackgroundColor` is the default color.

Note that when the page is scrolled, and the Navigation Bar is reduced, the correct color is displayed.

### Basic Information

- Version with issue: I tested from v4.3.0.819712-pre2 back to Xamarin.Forms v3.5.0.274416 and confirmed the same issue persists (My hunch is that this is a breaking change introduced in iOS 13)
- Last known good version: None (My hunch is that this is a breaking change introduced in iOS 13)
- IDE: Visual Studio Mac
- Platform Target Frameworks: 
  - iOS:  Xamarin.iOS v13.2.0.42

### Reproduction Link

https://github.com/brminnick/iOSLargeTitleRepro

### PR DESCRIPTION
**Pull request by adrianknight89**
### Description of Change ###

Incorporated changes in iOS13 pertaining to navigation bar background color.

### Issues Resolved ### 

- fixes #7701
- fixes #7350 
- fixes #7679 (possibly)

### PR Checklist ###

- [x] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Platform.iOS/Renderers/NavigationRenderer.cs [hunk 1]
```diff
@@ -622,45 +617,65 @@ void UpdateBackgroundColor()
...
-					: barTextColor.ToUIColor();
-				NavigationBar.TitleTextAttributes = titleAttributes;
-			}
+				if (barTextColor == Color.Default)
+				{
+					if (NavigationBar.TitleTextAttributes != null)
+						NavigationBar.TitleTextAttributes = titleTextAttributes;
+				}
+				else
+					NavigationBar.TitleTextAttributes = titleTextAttributes;
```
Comment (adrianknight89) [inline Xamarin.Forms.Platform.iOS/Renderers/NavigationRenderer.cs]: Lines 669 - 675 can possibly be simplified by `NavigationBar.TitleTextAttributes = titleTextAttributes;`, but I'm not sure why the previous code had that extra null check on `NavigationBar.TitleTextAttributes`.

**Comment by adrianknight89** [comment]
@samhouts Which branch should this be targeted for?

**Comment by jfversluis** [comment]
@adrianknight89 since it's regression it should be 4.2 right now.

Does this change use any iOS13/Xcode 11 specific APIs? If so, this can't be merged before our build pipeline is upgraded to that.

**Comment by adrianknight89** [comment]
@jfversluis Yes, it does use iOS13 APIs. When will the build pipeline be upgraded?

**Comment by jfversluis** [comment]
Unfortunately that is a bit of a problem at the moment... #7894

**Comment by adrianknight89** [comment]
This seems like an urgent issue to fix, no? How long before we can move to the 2019 host?

**Comment by adrianknight89** [comment]
@samhouts changed the target branch

**Comment by samhouts** [comment]
@adrianknight89 We're working on getting out build compatible with both 2019 and 2017, and we'll hopefully have something working in the next few weeks. Thanks!

**Comment by rmarinho** [comment]
Hey @adrianknight89  can you rebase this one so we can run on vs2019? 

Thanks

**Comment by rmarinho** [comment]
opened #9346 rebased

**Pull request closed without merging**